### DISCUSSION TITLE
Consider adding ability to import various text files

### DISCUSSION THREAD

**Discussion by 66tree** [usememos/memos#3102] [Ideas]
Adding the ability to import .txt, .md and especially .org files would make this a killer option in my opinion.

**Comment by MehadND** [comment]
You mean like drag n drop feature? Should the text be added or just files as embedded content?

And also can you possibly give a use case or example?

**Comment by 66tree** [comment]
Thinking it could flow like this: you drag a .txt, .md or .org file into the area where it says "Any thoughts..." and it lets you preview the text you're about to post before hitting the save button. If you've seen how reddit allows you to preview your markdown before posting that's more or less what I'm visualizing but allowing you to drag and drop a few basic formats.  It would allow users to quickly convert text notes to Memos.

**Reply by MehadND** [reply]
Ah ok so kind of like drag n drop feature for pasting text into the editor

**Reply by 66tree** [reply]
Yeah that would be very nice to have. Then when the text is previewed before posting it would keep the format of those files or convert them to Markdown. 
I don't know if you're familiar with Emacs or [Org Mode](https://orgmode.org/) but Org Mode is like this supercharged outliner that is pretty much the only reason why I touch Emacs these days. It's amazing way to organize information and I would love to find some of it's functionality in this software. Just something to consider. At the end of the day it's all just formatted text and being able to import and export it from Memos would be a killer feature in my opinion.

**Reply by MehadND** [reply]
Yeah not really sure about emacs or org mode but will have a look at it soon

**Comment by 66tree** [comment]
I was looking at [memos content syntax](https://www.usememos.com/docs/getting-started/content-syntax) and Memos is already supporting much of the same functionality as Org Mode. A very basic introduction to the syntax is shown [here](https://orgmode.org/quickstart.html). Seems like conversion to Memos Markdown when using drag and drop into Memos would be doable. Emacs Org Mode has a lot of functionality that just doesn't apply to Memos use case, but there's a subset of near-identical syntax there that seems like low hanging fruit for this project. [Tables](https://orgmode.org/features.html) are implemented in almost the same manner for instance. Even if this functionality doesn't get implemented in Memos, there's a lot of ideas in Org Mode that might be great for future releases of Memos.
